src/main/java/tr/com/srdc/cda2fhir/transform/ResourceTransformerImpl.java

```diff
@@ -1991,22 +2020,12 @@ public EntryResult tMedicationDispense2MedicationDispense(
 
 		// product.manufacturedProduct(MedicationInformation) -> medication
 		if (cdaMedicationDispense.getProduct() != null && !cdaMedicationDispense.getProduct().isSetNullFlavor()) {
-			if (cdaMedicationDispense.getProduct().getManufacturedProduct() != null
-					&& !cdaMedicationDispense.getProduct().getManufacturedProduct().isSetNullFlavor()) {
-				if (cdaMedicationDispense.getProduct().getManufacturedProduct() instanceof MedicationInformation) {
-					MedicationInformation cdaMedicationInformation = (MedicationInformation) cdaMedicationDispense
-							.getProduct().getManufacturedProduct();
-					Medication fhirMedication = null;
-					Bundle fhirMedicationBundle = tMedicationInformation2Medication(cdaMedicationInformation);
-
-					for (BundleEntryComponent entry : fhirMedicationBundle.getEntry()) {
-						result.addResource(entry.getResource());
-						if (entry.getResource() instanceof org.hl7.fhir.dstu3.model.Medication) {
-							fhirMedication = (org.hl7.fhir.dstu3.model.Medication) entry.getResource();
-						}
-					}
-					fhirMediDisp.setMedication(new Reference(fhirMedication.getId()));
-				}
+			EntryResult fhirMedicationResult = tManufacturedProduct2Medication(cdaMedicationDispense.getProduct());
+			if (fhirMedicationResult.hasResult()) {
+				// Should only have one result entry
+				BundleEntryComponent medicationResultEntry = fhirMedicationResult.getBundle().getEntryFirstRep();
+				result.addResource(medicationResultEntry.getResource());
+				fhirMediDisp.setMedication(new Reference(medicationResultEntry.getId()));
```

Comment:
it's true, but you took off the instanceof type checking by doing it this way. Was this intentional?

Reply:
When I looked at what the tMedicationInformation2Medication function did, it just rerouted to the 2ManufacturedProduct2Medication so I thought it was a bit redundant. Maybe the instanceOf check is good for another reason?

Reply:
If it's possible to have a manufacturedProduct inside of medicalDispense without it being a MedicationInformation then I see the need for it.

Reply:
The spec does say it could be either a ImmunizationMedicationInformation or a MedicationInformation. Do we only want to include one or both?

Reply:
steven to add .hasResource() check on ~ 2027

Reply:
can add med and imm type checking too.